qiskit/pulse/instructions/directives.py

```diff
@@ -52,3 +55,66 @@ def __eq__(self, other):
         """Verify two barriers are equivalent."""
         return (isinstance(other, type(self)) and
                 set(self.channels) == set(other.channels))
+
+
+class Call(Directive):
+    """Pulse ``Call`` directive.
+
+    This instruction wraps another program when ``pulse.call`` function is
+    used in the pulse builder context. Note that this instruction is not exposed to users.
+    This is implicitly applied to create a better target code in the Qiskit compiler.
+    This instruction clearly indicates the attached schedule is a subroutine
+    that is defined outside of the current scope, i.e. no need to redefine programs
+    within the current scope.
+
+    The metadata attached to the subroutine is kept.
+    """
+
+    # note that we cannot type hint for this due to cyclic import
+    def __init__(self, subroutine, name: Optional[str] = None):
+        """Create a new call directive with subroutine.
+
+        Note that no further optimization or transformation are performed between
+        subroutines because they are externally defined programs.
+        However we can assign arbitrary parameter to subroutines,
+        given these parameter values are managed by unique parameter objects.
+
+        Args:
+            subroutine (Schedule): A program to wrap with call instruction.
+        """
+        if name is None:
+            routine_id = hashlib.md5(str(subroutine.instructions).encode('utf-8')).hexdigest()
+        else:
+            routine_id = name
+
+        super().__init__((subroutine,), None,
+                         channels=tuple(subroutine.channels),
+                         name=routine_id)
+
```

Comment:
Just an option. Not a strong suggestion
```suggestion
        """
        if name is None:
            name = hashlib.md5(str(subroutine.instructions).encode('utf-8')).hexdigest()

        super().__init__((subroutine,), None,
                         channels=tuple(subroutine.channels),
                         name=name)

```